clustering.Rmd

```diff
@@ -45,95 +49,147 @@ use the subgroups to generate new questions about the data and follow up with a
 way.
-```{r 09-penguins, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 3, fig.width = 4, fig.cap = "Gentoo penguin. [@penguinsimage]", fig.retina = 2}
-knitr::include_graphics("img/gentoo.jpg")
+
+```{r 09-penguins, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 3, fig.width = 4, fig.cap = "Gentoo penguin.", fig.retina = 2}
+knitr::include_graphics("img/Gentoo_Penguin._(8671680772).jpg")
 ```
 
-Below we will work with `penguin_data`, a subset of 18 observations of the original data, which has already been scaled. We will discuss scaling for K-means in more detail later in this chapter. \index{mutate}\index{read function!read\_csv} 
+To learn about K-means clustering in this chapter, 
+we will work with `penguin_data`, 
+a subset of 18 observations of the original data, 
+which has already been standardized 
+(remember from Chapter \@ref(classification) that scaling is part of the standardization process). 
+We will discuss scaling for K-means in more detail later in this chapter. 
+\index{mutate}\index{read function!read\_csv} 
+
+Before we get started, we will load the `tidyverse` metapackage to ensure
+we have the functions we need, 
+as well as set a random seed, so that our analysis will be reproducible.
+As we will learn in more detail later in the chapter, 
+the K-means clustering algorithm uses random numbers - 
+which is why setting the seed here is important.
+
 ```{r 10-toy-example-data, echo = FALSE, message = FALSE, warning = FALSE}
 library(tidyverse)
 
 data <- read_csv("data/toy_penguins.csv") |>
   mutate(cluster = as_factor(cluster)) |>
-  mutate(flipper_length_scaled = scale(flipper_length_mm), bill_length_scaled = scale(bill_length_mm))
+  mutate(flipper_length_standardized = as.double(scale(flipper_length_mm)), 
+         bill_length_standardized = as.double(scale(bill_length_mm)))
 
-penguin_data <- data |> select(flipper_length_scaled, bill_length_scaled)
+penguin_data <- data |> select(flipper_length_standardized, bill_length_standardized)
+write_csv(penguin_data, "data/penguins_standardized.csv")
 ```
 
 \index{seed!set.seed}
 
-```{r 10-clustering, warning = F, message=F}
+```{r 10-clustering, warning = FALSE, message = FALSE}
 library(tidyverse)
-
 set.seed(1)
+```
 
+Now we can load and preview the data.
+
+```{r, message = FALSE, warning = FALSE}
+penguin_data <- read_csv("data/penguins_standardized.csv")
 penguin_data
 ```
 
-```{r 10-toy-example-plot, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Subset data of penguin scaled bill length versus flipper length."}
-base <- ggplot(data, aes(x = flipper_length_scaled, y = bill_length_scaled)) +
+Next, we can create a scatter plot using this data set to see if we can use
+data visualization to detect subtypes or groups in our data set.
+
+```{r 10-toy-example-plot, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Subset data of penguin standardized bill length versus standardized flipper length."}
```

Comment:
"Subset data of penguin standardized bill length" -- this is very unclear

Reply:
I changed it to just: "Scatter plot of standardized bill length versus standardized flipper length."